13_iv_antibiotics_pharmacy.sql

```diff
@@ -0,0 +1,87 @@
+intime,
+outtime,
+medication,
+min(entertime) as entertime,
+min(starttime) as starttime,
+min(route) as route,
+max(stoptime) as stoptime,
+min(frequency) as frequency,
+min(disp_sched) as disp_sched,
+min(doses_per_24_hrs) as doses_per_24_hrs,
+min(duration) as duration,
+min(duration_interval) as duration_interval,
+count(*) as records
+from `physionet-data.mimic_hosp.pharmacy`
+JOIN `physionet-data.mimic_icu.icustays`
+USING (subject_id, hadm_id)
+where
+(
+    --medical abx names
+    lower(medication) like "%vancomycin%"
+    or lower(medication) like "%ceftriaxone%"
+    or lower(medication) like "%meropenem%"
+    or lower(medication) like "%ceftazidime%"
+    or lower(medication) like "%cefotaxime%"
+    or lower(medication) like "%cefepime%"
+    or lower(medication) like "%piperacillin%"
+    or lower(medication) like "%tazobactam%"
+    or lower(medication) like "%amipcillin%"
+    or lower(medication) like "%sulbactam%"
+    or lower(medication) like "%imipenem%"
+    or lower(medication) like "%cilastatin%"
+    or lower(medication) like "%levofloxacin%"
+    or lower(medication) like "%clindamycin%"
+    or lower(medication) like "%metronidazol%"
+    or lower(medication) like "%linezolid%"
+    or lower(medication) like "%ciprofloxacin%"
+    or lower(medication) like "%nafcillin%"
+    or lower(medication) like "%daptomycin%"
+    or lower(medication) like "%amikacin%"
+    or lower(medication) like "%tobramycin%"
+    or lower(medication) like "%gentamicin"
+    --brand names
+    or lower(medication) like "%vancocin%"
+    or lower(medication) like "%rocephin%"
+    or lower(medication) like "%merrem%"
+    or lower(medication) like "%fortaz%"
+    or lower(medication) like "%claforan%"
+    or lower(medication) like "%maxipime%"
+    or lower(medication) like "%zosyn%"
+    or lower(medication) like "%unasyn%"
+    or lower(medication) like "%primaxin%"
+    or lower(medication) like "%levaquin%"
+    or lower(medication) like "%cleocin%"
+    or lower(medication) like "%flagyl%"
+    or lower(medication) like "%zyvox%"
+    or lower(medication) like "%cetraxal%"
+    or lower(medication) like "%cipro%"
+    or lower(medication) like "%otiprio%"
+    or lower(medication) like "%ciloxan%"
+    or lower(medication) like "%cubicin%"
+    or lower(medication) like "%tobi podhaler%"
+    or lower(medication) like "%bethkis%"
+    or lower(medication) like "%tobrex%"
+    or lower(medication) like "%gentak%"
+)
+and route in (
+    'IV' --this captures by far the largest portion of abx orders, followed with a few from IVIC and IVT. The others are here for posterity
+    ,'IVCI'
+    ,'IVIC'
+    ,'IVS'
+    ,'IVT')
+AND
+lower(medication) not like "%desensitization%" --ignore desensitization protocols in preparation for abx
+AND
+DATETIME_TRUNC(stoptime,
+    HOUR) < outtime 
+AND
+DATETIME_TRUNC(starttime,
+    HOUR) > intime
+GROUP BY subject_id, hadm_id, stay_id, intime, outtime, medication, chart_hour
```

Comment:
I don't quite grok everything in the `GROUP BY` here for the aggregate stats:
- Is `stay_id` already globally unique? If so, then `subject_id` and `hadm_id` should already be the same if the `stay_id` is unique.
- If this is grouped around the stay, then should `intime` and `outtime` already correspond to it? What happens if the `intime` and `outtime` are not unique to a stay - is it an error?
- I don't think there's an easy way around binning by `chart_hour`, and it's probably fine for the foreseeable future. I get concerned if there are relatively short protocols where the amount of time truncated by `chart_hour` becomes significant, but for simplicity I think it's something we should live with. (Apologies for the tangent.) 
-

Reply:
Hi Scott, just got around to looking over this comment. I believe Stay_ID is the most fine-grained compared to subject and hadm_id - I think we're including the subject and hadm_id for future examination, like to identify several stays that were actually the same patient in the same hospital admission.

As for intime and outtime, those correspond to a stay_id - those are only in the select and groupby clauses because I wanted to visually be able to check and compare our medication records' entertime as well as starttime/endtime relative to the stay_id's intime and outtime - for example, some medications are ordered with a starttime that is within the stay's in and outtime, but the endtime is ordered for a time later than this range. I think we can take it out in the final PR change as this was mainly for my benefit while developing the query.

Reply:
Ok, sounds good. It sounds like grouping by `stay_id` and `medication` is the right way to go if I've understood correctly.